teamfoundationcontrol.cpp

```diff
@@ -231,3 +249,36 @@ Core::ShellCommand *TeamFoundation::Internal::TeamFoundationControl::createIniti
     command->addJob(settings.binaryPath(), getArgs, -1);
     return command;
 }
+
+////////////////////////////////////////////////////////////////////////////////
+void TeamFoundationControl::DirectoryTree::Add(const QString& directory)
+{
+    for (int i = 0; i < m_entries.size(); ++i) {
+        // candidate shadowed?
+        if (IsParentOrSame(m_entries[i], directory)) {
+            return;
+        }
+
+        // candidate shadows?
+        if (IsParentOrSame(directory, m_entries[i])) {
+            m_entries[i] = m_entries[m_entries.size()-1];
+            m_entries.pop_back();
+            --i;
+        }
+    }
+
+    m_entries << directory;
+}
+
+bool TeamFoundationControl::DirectoryTree::IsParentOrSame(const QString& lsh, const QString& rhs)
+{
+    return rhs.startsWith(lsh, Qt::CaseInsensitive);

```

Comment:
Does Qt Creator guarantee that this is always called with a trailing slash? Otherwise I assume it could give false positives (where two sibling directories have a common prefix).